authors/apps/profiles/views.py

```diff
@@ -46,3 +68,101 @@ def update(self, request, username):
             serializer.save()
             return Response(serializer.data)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class FollowUnfollow(CreateAPIView):
+    permision_classes = (IsAuthenticated,)
+    serializer_class = ProfileSerializer
+
+    def post(self, request, username):
+        """
+        This method handles following a user
+        """
+        # get current user
+        following_user = check_user_profile(request)
+        # check if the user to be followed exists
+        try:
+            followed_user = followed_user_profile(username)
+        except Profile.DoesNotExist:
+            raise NotFound(
+                'You are trying to follow a user who doe not exist, please check the username again')
+        # Check if user is trying to follow himself
+        if following_user.id == followed_user.id:
+            raise ValidationError("You cannot follow yourself")
+        # Add user
+        following_user.follow(followed_user)
+
+        serialize = self.serializer_class(
+            followed_user, context={'request': request})
+        return Response(data=serialize.data, status=status.HTTP_201_CREATED)
+
+    def delete(self, request, username):
+        """
+        This method handles unfollowing a user
+        """
+        # get current user
+        following_user = check_user_profile(request)
+        
+        # check if the user to unfollowed be exists
+        followed_user = followed_user_profile(username)
+        if not followed_user:
+            raise NotFound(
+                'You are trying to unfollow a user who in not there,please check the username again')
+
+        # Check if user trying to unfollow him/herself
+        if following_user.id == followed_user.id:
+            raise ValidationError("You cannot unfollow yourself")
+
+        # unfollow user
+        following_user.unfollow(followed_user)
+
+        serialize = self.serializer_class(
+            followed_user, context={'request': request})
+        return Response(data=serialize.data, status=status.HTTP_200_OK)
+
+
+class Following(RetrieveAPIView):
+    """
+    Get all the users a user follows
+    """
+    permission_classes = (IsAuthenticated,)
+    serializer_class = ProfileSerializer
+
+    def get(self, request, username):
+        """Following"""
+
+        following_user = current_user_profile(request)
+        try:
+            followed_user = followed_user_profile(username)
+        except Exception:
+            raise NotFound(
+                'No such user found,please check the username again')
```

Comment:
Don't just test the status code,Test these messages too to increase test coverage